### PR TITLE
feature/tsconfig/export

### PR DESCRIPTION
**Pull request by bonzzy**
Prepared `tsconfig` standard for easy usage in projects

### CODE REVIEW

File: docs/typescript/tsconfig/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
+    "typeRoots": ["node_modules/@types", "src/types"], /* List of folders to include type definitions from. */
+    //     "types": [
+    //       "@types", "src/typings"
+    //     ],                           /* Type declaration files to be included in compilation. */
+    "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
+
+    /* Source Map Options */
+    "sourceRoot": "src/ts",                    /* Specify the location where debugger should locate TypeScript files instead of source locations. */
```
Comment (tonymrakovcic) [inline docs/typescript/tsconfig/tsconfig.json]: shoudnt this be just src?

Comment (bonzzy) [reply docs/typescript/tsconfig/tsconfig.json]: yes I think so too 👍

File: docs/typescript/tsconfig/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    // "noUnusedParameters": true,            /* Report errors on unused parameters. */
+    // "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
+    // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
+
+    /* Module Resolution Options */
+    "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+    // "baseUrl": "./",                       /* Base directory to resolve non-absolute module names. */
+    // "paths": {},                           /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+    // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
+    "typeRoots": ["node_modules/@types", "src/types"], /* List of folders to include type definitions from. */
```
Comment (vilimco) [inline docs/typescript/tsconfig/tsconfig.json]: ne treba node_modules, ```@types``` se resolvea u ```node_modules/@types```.

Nisam siguran, ali generalno postoji mogucnost da se node_modules ne zove node_modules, ovako imamo pokriveno uvijek :)

File: docs/typescript/tsconfig/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    //     "types": [
+    //       "@types", "src/typings"
+    //     ],                           /* Type declaration files to be included in compilation. */
+    "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
+    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
+
+    /* Source Map Options */
+    "sourceRoot": "src/",                    /* Specify the location where debugger should locate TypeScript files instead of source locations. */
```
Comment (vilimco) [inline docs/typescript/tsconfig/tsconfig.json]: We have trailing slash because?
Remove it.

**Review by vilimco — CHANGES_REQUESTED** [review]

File: docs/typescript/tsconfig/tsconfig.json (hunk 2 above)
Comment (tonymrakovcic) [reply docs/typescript/tsconfig/tsconfig.json]: ne znam dal mislis na ovo - https://github.com/npm/npm/issues/2734#issuecomment-7919638
Ali s njihove strane se sam string "node_modules" nece mijenjat :)
Samo digresija, kuzim kaj mislis inace :)

Comment (krukru) [reply docs/typescript/tsconfig/tsconfig.json]: Meni se ne kompajliraju testovi ako imam `@types`, dok `node_modules/@types` radi

Comment (bonzzy) [reply docs/typescript/tsconfig/tsconfig.json]: meni također radi samo dok je `node_modules/@types`

Comment (vilimco) [reply docs/typescript/tsconfig/tsconfig.json]: hmm, dost zanimljivo. Nek ostane onda ovako.

Meni radi samo ```@types```, cak i za jasmine testove koje sam vrtio na onom svom setupu :/

**Review by vilimco — APPROVED** [review]

**Comment by krukru** [comment]
Same thing as with tslint.json, the config file is in docs folder. Other than that, I see that @jewbre approved so I will assume the doc is final (for now)

**Pull request merged**